### DISCUSSION TITLE
How to use form actions within DropdownMenu

### DISCUSSION THREAD

**Discussion by azezsan** [huntabyte/shadcn-svelte#793] [Help]
I'm trying to make pressing delete within the dropdown submit a form action to the server to delete the row but I didn't know how to implement it well, I managed to get it working but I don't like my solution, I would like a more simple solution that doesn't use javascript, as well I would like to know if I'm not using the API correctly...

```
<script lang="ts">
	import DotsHorizontal from 'svelte-radix/DotsHorizontal.svelte';
	import { Button } from '$lib/components/ui/button';
	import * as DropdownMenu from '$lib/components/ui/dropdown-menu';
	import { labels } from '../(data)/data';
	import { taskSchema, type Task } from '../(data)/schemas';
	import { enhance } from '$app/forms';

	export let row: Task;
	const task = taskSchema.parse(row);

	let form: HTMLFormElement;
</script>

<DropdownMenu.Root>
	<DropdownMenu.Trigger asChild let:builder>
		<Button
			variant="ghost"
			builders={[builder]}
			class="flex h-8 w-8 p-0 data-[state=open]:bg-muted"
		>
			<DotsHorizontal class="h-4 w-4" />
			<span class="sr-only">Open menu</span>
		</Button>
	</DropdownMenu.Trigger>
	<DropdownMenu.Content class="w-[160px]" align="end">
		<DropdownMenu.Item>Edit</DropdownMenu.Item>
		<DropdownMenu.Item>Make a copy</DropdownMenu.Item>
		<DropdownMenu.Item>Favorite</DropdownMenu.Item>
		<DropdownMenu.Separator />
		<DropdownMenu.Sub>
			<DropdownMenu.SubTrigger>Labels</DropdownMenu.SubTrigger>
			<DropdownMenu.SubContent>
				<DropdownMenu.RadioGroup value={task.label}>
					{#each labels as label}
						<DropdownMenu.RadioItem value={label.value}>
							{label.label}
						</DropdownMenu.RadioItem>
					{/each}
				</DropdownMenu.RadioGroup>
			</DropdownMenu.SubContent>
		</DropdownMenu.Sub>
		<DropdownMenu.Separator />
		<form action="?/delete" method="POST" use:enhance bind:this={form}>
			<DropdownMenu.Item on:click={() => form.submit()}>
				<input type="hidden" name="id" value={task.id} />
				Delete
				<DropdownMenu.Shortcut>⌘⌫</DropdownMenu.Shortcut>
			</DropdownMenu.Item>
		</form>
	</DropdownMenu.Content>
</DropdownMenu.Root>
```

![image](https://github.com/huntabyte/shadcn-svelte/assets/79533966/0f28a78b-7f0e-4307-ab5b-9d6146d43185)

**Comment by azezsan** [comment]
I realized doing it with javascript makes me unable to enhance the form, need help 🥺

**Reply by shyakadavis** [reply]
Hi;

Given recent updates, a lot needs to be changed, but you could refer to this conversation (https://github.com/huntabyte/shadcn-svelte/discussions/679) for a similar solution.

I think the same principles apply.

Good luck. 🙂

**Comment by brucegl** [comment]
Hi,

This pattern worked for me

```
<DropdownMenu.Group>
	<DropdownMenu.Item>
		<Avatar />
		Profile
	</DropdownMenu.Item>
	<DropdownMenu.Item>
		<LockClosed />
		Security
	</DropdownMenu.Item>
	<form method="post" action="?/logout" use:enhance bind:this={form}>
		<DropdownMenu.Item on:click={() => form.submit()}>
			<Exit />
			<input type="submit" value="Sign out" />
		</DropdownMenu.Item>
	</form>
</DropdownMenu.Group>
```
Hope that helps someone

**Reply by Wizzel1** [reply]
@brucegl 
Does enhance have any effect for you? In my case the page still reloads and I get the query appended to the url

**Reply by brucegl** [reply]
@Wizzel1 , sorry for the delay (working on IoT cloud infrastructure). I tried it with and without use:enhance, and with and without JS in the browser, and I didn't notice any difference.

**Reply by Wizzel1** [reply]
@brucegl No worries, thanks for the reply. Guess i need to give this another try

**Reply by slingshotsys** [reply]
> Does enhance have any effect for you? In my case the page still reloads and I get the query appended to the url

@Wizzel1 did you ever find a solution for getting `use:enhance` to work? I'm bumping into the same issue.

**Reply by slingshotsys** [reply]
Nevermind, figured it out.  Just use `form.requestSubmit();` instead of form.submit() 🤦

**Comment by devjume** [comment]
Just to point out: If you problem where form is automatically highlighted/focused when it is inside `<DropdownMenu.Item>`.
Add `tabindex='-1'` attribute to the `<button>` or `<input>` element.
Check my comment about this: https://github.com/huntabyte/shadcn-svelte/discussions/934#discussioncomment-8888624
